From e553c841a539e9f4e3847ca3467e4ce92107e394 Mon Sep 17 00:00:00 2001
From: Tobias Schramm <t.schramm@t-sys.eu>
Date: Fri, 9 Apr 2021 21:15:29 +0200
Subject: [PATCH 16/16] backlight: always parse default-brightness-level
 property

Previously the default-brightness-level property was only parsed if
the brightness-levels property is set, too. Therefore there is no
method to specify the desired default brightness in a device tree
without also specifying  a brightness-levels table, too. Since this
is discouraged this commit changes that behaviour to parse any present
default-brightness-level property, even if there is no brightness-levels
property.

Signed-off-by: Tobias Schramm <t.schramm@t-sys.eu>
---
 drivers/video/backlight/pwm_bl.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/video/backlight/pwm_bl.c b/drivers/video/backlight/pwm_bl.c
index dfc760830eb9..9381b8d575e1 100644
--- a/drivers/video/backlight/pwm_bl.c
+++ b/drivers/video/backlight/pwm_bl.c
@@ -616,6 +616,18 @@ static int pwm_backlight_probe(struct platform_device *pdev)
 		goto err_alloc;
 	}
 
+	if (device_property_present(&pdev->dev, "default-brightness-level")) {
+		uint32_t value;
+
+		ret = device_property_read_u32(&pdev->dev,
+					       "default-brightness-level",
+					       &value);
+		if (ret < 0)
+			goto err_alloc;
+
+		data->dft_brightness = value;
+	}
+
 	if (data->dft_brightness > data->max_brightness) {
 		dev_warn(&pdev->dev,
 			 "invalid default brightness level: %u, using %u\n",
-- 
2.30.1

